Sources/SecureEnclaveValet.swift

```diff
@@ -0,0 +1,285 @@
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+
+
+/// Reads and writes keychain elements that are stored on the Secure Enclave using Accessibility attribute `.whenPasscodeSetThisDeviceOnly`. Accessing these keychain elements will require the user to confirm their presence via Touch ID, Face ID, or passcode entry. If no passcode is set on the device, accessing the keychain via a `SecureEnclaveValet` will fail. Data is removed from the Secure Enclave when the user removes a passcode from the device.
+@objc(VALSecureEnclaveValet)
+public final class SecureEnclaveValet: NSObject {
+    
+    // MARK: Public Class Methods
+    
+    /// - parameter identifier: A non-empty string that uniquely identifies a SecureEnclaveValet.
+    /// - returns: A SecureEnclaveValet that reads/writes keychain elements with the desired flavor.
+    public class func valet(with identifier: Identifier, accessControl: SecureEnclaveAccessControl) -> SecureEnclaveValet {
+        let key = Service.standard(identifier, .secureEnclave(accessControl)).description as NSString
+        if let existingValet = identifierToValetMap.object(forKey: key) {
+            return existingValet
+            
+        } else {
+            let valet = SecureEnclaveValet(identifier: identifier, accessControl: accessControl)
+            identifierToValetMap.setObject(valet, forKey: key)
+            return valet
+        }
+    }
+    
+    /// - parameter identifier: A non-empty string that must correspond with the value for keychain-access-groups in your Entitlements file.
+    /// - returns: A SecureEnclaveValet that reads/writes keychain elements that can be shared across applications written by the same development team.
+    public class func sharedAccessGroupValet(with identifier: Identifier, accessControl: SecureEnclaveAccessControl) -> SecureEnclaveValet {
+        let key = Service.sharedAccessGroup(identifier, .secureEnclave(accessControl)).description as NSString
+        if let existingValet = identifierToValetMap.object(forKey: key) {
+            return existingValet
+            
+        } else {
+            let valet = SecureEnclaveValet(sharedAccess: identifier, accessControl: accessControl)
+            identifierToValetMap.setObject(valet, forKey: key)
+            return valet
+        }
+    }
+    
+    // MARK: Equatable
+    
+    /// - returns: `true` if lhs and rhs both read from and write to the same sandbox within the keychain.
+    public static func ==(lhs: SecureEnclaveValet, rhs: SecureEnclaveValet) -> Bool {
+        return lhs.service == rhs.service
+    }
+    
+    // MARK: Private Class Properties
+    
+    private static let identifierToValetMap = NSMapTable<NSString, SecureEnclaveValet>.strongToWeakObjects()
+    
+    // MARK: Initialization
+    
+    @available(*, deprecated)
+    public override init() {
+        fatalError("Do not use this initializer")
```

Comment:
This is a very vague error.  Should specify which methods to use to initialize instead.

Reply:
Also applies to `Valet` and `SinglePromptSecureEnclaveValet`

Reply:
fair. Will also add `@available(swift, obsoleted: 1.0)` to these methods to prevent them from being called from Swift.